src/Bool.hs

```diff
@@ -52,8 +59,32 @@ unlessM p m = p >>= flip unless m
 -- True text
 ifM :: Monad m => m Bool -> m a -> m a -> m a
 ifM p x y = p >>= \b -> if b then x else y
+{-# INLINE ifM #-}
 
-{-
+-- | Version of 'guard' that takes verification function.
+-- Can be used in some similar way:
+-- @
+--     safeSum :: Int -> Int -> Maybe Int
+--     safeSum a b = do
+--         verifiedA <- guarded (>0) a
+--         verifiedB <- guarded (>0) b
+--         pure $ verifiedA + verifiedB
+-- @
+guarded :: Alternative f => (a -> Bool) -> a -> f a
```

Comment:
What motivates this addition? I see no advantage of `guarded (>0) a` over `guard (a > 0)` in the provided example.

Reply:
Nothing. Now I can see that there's really not much sense in having `guarded` function. I used it several times. But now I don't use it. I think it can be safely removed.